src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Flow.java

```diff
@@ -743,7 +743,7 @@ public void visitSwitchExpression(JCSwitchExpression tree) {
         }
 
         sealed interface PatternDescription {
-            public static PatternDescription from(Types types, Type selectorType, JCPattern pattern) {
+            public static PatternDescription from(Types types, Type selectorType, JCPattern pattern, Symtab syms) {
```

Comment:
Optional comment: the fact that this factory needs so much compiler state makes me think that perhaps a better place for it would be as an instance method of Flow, rather than a static method in PatternDescription.

Reply:
@lahodaj WDYT? Looks like it can be easily refactored and it is used only in one place (apart from the recursive call). I can refactor it in this PR or u can do it in #13074.